### ISSUE TITLE
What is the difference between ''optimizer_start'' and ''optimizer''?

### ISSUE THREAD

**Issue by EricKani** [Haochen-Wang409/U2PL#56]
Like the title~ They are defined in the file "train_semi.py".
Thanks for your great work! It gives me some insights for my project~

Best,
Eric

**Comment by Haochen-Wang409** [comment]
Thanks for your attention of our work!

```optimizer_start``` is used to compute the learning rate when we try to **resume** a model, while ```optimizer``` is used to train the model.

**Comment by EricKani** [comment]
Thanks for your quick response! And sorry for the late reply...

I also don't understand the reason why you set 'optimizer_start' here... When we implement 'lr_scheduler.step()', the lr in optimizer_start will be updated, but not optimizer. 
When we use optimizer to train the model, will the lr always be the init lr?

Thanks!
Eric

**Comment by Haochen-Wang409** [comment]
In fact, we don't know whether the ```lr``` will remain the initial value during training.

The reason why we adopt an ```optimizer_start``` is for fix wrong lr when **resume** a model, see https://github.com/Haochen-Wang409/U2PL/blob/main/u2pl/utils/lr_helper.py#L56-L58

The ```base_lr``` of ```lr_scheduler``` is computed by the learning rate of optimizer. However, the ```base_lr``` should be always a pre-defined constant. That's why we use an extra optimizer to get initial lr.
